### ISSUE TITLE
Vulnerability roundup 111: onionshare-2.4: 9 advisories [7.5]

### ISSUE THREAD

**Issue by ckauhaus** [NixOS/nixpkgs#160688]
[search](https://search.nix.gsc.io/?q=onionshare&i=fosho&repos=NixOS-nixpkgs), [files](https://github.com/NixOS/nixpkgs/search?utf8=%E2%9C%93&q=onionshare+in%3Apath&type=Code)

* [ ] [CVE-2022-21689](https://nvd.nist.gov/vuln/detail/CVE-2022-21689) CVSSv3=7.5 (nixos-unstable)
* [ ] [CVE-2022-21693](https://nvd.nist.gov/vuln/detail/CVE-2022-21693) CVSSv3=6.5 (nixos-unstable)
* [ ] [CVE-2022-21688](https://nvd.nist.gov/vuln/detail/CVE-2022-21688) CVSSv3=5.5 (nixos-unstable)
* [ ] [CVE-2022-21690](https://nvd.nist.gov/vuln/detail/CVE-2022-21690) CVSSv3=5.4 (nixos-unstable)
* [ ] [CVE-2022-21694](https://nvd.nist.gov/vuln/detail/CVE-2022-21694) CVSSv3=5.3 (nixos-unstable)
* [ ] [CVE-2022-21695](https://nvd.nist.gov/vuln/detail/CVE-2022-21695) CVSSv3=5.3 (nixos-unstable)
* [ ] [CVE-2022-21691](https://nvd.nist.gov/vuln/detail/CVE-2022-21691) CVSSv3=4.3 (nixos-unstable)
* [ ] [CVE-2022-21692](https://nvd.nist.gov/vuln/detail/CVE-2022-21692) CVSSv3=4.3 (nixos-unstable)
* [ ] [CVE-2022-21696](https://nvd.nist.gov/vuln/detail/CVE-2022-21696) CVSSv3=4.3 (nixos-unstable)

### CVE-2022-21689

OnionShare is an open source tool that lets you securely and anonymously share files, host websites, and chat with friends using the Tor network. In affected versions the receive mode limits concurrent uploads to 100 per second and blocks other uploads in the same second, which can be triggered by a simple script. An adversary with access to the receive mode can block file upload for others. There is no way to block this attack in public mode due to the anonymity properties of the tor network.

### CVE-2022-21693

OnionShare is an open source tool that lets you securely and anonymously share files, host websites, and chat with friends using the Tor network. In affected versions an adversary with a primitive that allows for filesystem access from the context of the Onionshare process can access sensitive files in the entire user home folder. This could lead to the leaking of sensitive data. Due to the automatic exclusion of hidden folders, the impact is reduced. This can be mitigated by usage of the flatpak release.

### CVE-2022-21688

OnionShare is an open source tool that lets you securely and anonymously share files, host websites, and chat with friends using the Tor network. Affected versions of the desktop application were found to be vulnerable to denial of service via an undisclosed vulnerability in the QT image parsing. Roughly 20 bytes lead to 2GB memory consumption and this can be triggered multiple times. To be abused, this vulnerability requires rendering in the history tab, so some user interaction is required. An adversary with knowledge of the Onion service address in public mode or with authentication in private mode can perform a Denial of Service attack, which quickly results in out-of-memory for the server. This requires the desktop application with rendered history, therefore the impact is only elevated. This issue has been patched in version 2.5.

### CVE-2022-21690

OnionShare is an open source tool that lets you securely and anonymously share files, host websites, and chat with friends using the Tor network. In affected versions The path parameter of the requested URL is not sanitized before being passed to the QT frontend. This path is used in all components for displaying the server access history. This leads to a rendered HTML4 Subset (QT RichText editor) in the Onionshare frontend.

### CVE-2022-21694

OnionShare is an open source tool that lets you securely and anonymously share files, host websites, and chat with friends using the Tor network. The website mode of the onionshare allows to use a hardened CSP, which will block any scripts and external resources. It is not possible to configure this CSP for individual pages and therefore the security enhancement cannot be used for websites using javascript or external resources like fonts or images.

### CVE-2022-21695

OnionShare is an open source tool that lets you securely and anonymously share files, host websites, and chat with friends using the Tor network. In affected versions authenticated users (or unauthenticated in public mode) can send messages without being visible in the list of chat participants. This issue has been resolved in version 2.5.

### CVE-2022-21691

OnionShare is an open source tool that lets you securely and anonymously share files, host websites, and chat with friends using the Tor network. In affected versions chat participants can spoof their channel leave message, tricking others into assuming they left the chatroom.

### CVE-2022-21692

OnionShare is an open source tool that lets you securely and anonymously share files, host websites, and chat with friends using the Tor network. In affected versions anyone with access to the chat environment can write messages disguised as another chat participant.

### CVE-2022-21696

OnionShare is an open source tool that lets you securely and anonymously share files, host websites, and chat with friends using the Tor network. In affected versions it is possible to change the username to that of another chat participant with an additional space character at the end of the name string. An adversary with access to the chat environment can use the rename feature to impersonate other participants by adding whitespace characters at the end of the username.

-----
Scanned versions: nixos-unstable: 5aaed40d22f.

Cc @lourkeur

**Comment by lourkeur** [comment]
My understanding is that all of these vulnerabilities are patched in Onionshare 2.5, which is the version available in both nixos-21.11 and nixos-unstable.

```
$ nix eval nixpkgs/nixos-21.11#onionshare.version
"2.5"
$ nix eval nixpkgs/nixos-unstable#onionshare.version
"2.5"

**Comment by FliegendeWurst** [comment]
Yes, for some reason the scanned commit is a month old.
